src/ast/irbuilderbpf.cpp

```diff
@@ -603,62 +617,73 @@ CallInst *IRBuilderBPF::CreateGetRandom()
 
 CallInst *IRBuilderBPF::CreateGetStackId(Value *ctx, bool ustack, StackType stack_type)
 {
+  assert(ctx && ctx->getType() == getInt8PtrTy());
   assert(bpftrace_.stackid_maps_.count(stack_type) == 1);
-  Value *map_ptr = CreateBpfPseudoCall(bpftrace_.stackid_maps_[stack_type]->mapfd_);
+
+  Value *mapfd = CreateBpfPseudoCall(
+      bpftrace_.stackid_maps_[stack_type]->mapfd_);
+  assert(mapfd->getType()->isIntegerTy());
 
   int flags = 0;
   if (ustack)
     flags |= (1<<8);
   Value *flags_val = getInt64(flags);
 
-  // int bpf_get_stackid(ctx, map, flags)
+  // int bpf_get_stackid(struct pt_regs *ctx, struct bpf_map *map, u64
```

Comment:
Looks like the end of the comment got cut off

Reply:
Also if the 2nd argument is `struct bpf_map *` why's the type `Int64` below?

Reply:
I'm getting some serious deja vu about this question... sorry if you've already answered months ago on another pull request but I've forgotten :P

Reply:
The helper we have returns an int64:

```cpp
CallInst *IRBuilderBPF::CreateBpfPseudoCall(int mapfd)
{
  Function *pseudo_func = module_.getFunction("llvm.bpf.pseudo");
  return CreateCall(pseudo_func, {getInt64(BPF_PSEUDO_MAP_FD), getInt64(mapfd)}, "pseudo");
}
...
Value *map_ptr = CreateBpfPseudoCall(bpftrace_.stackid_maps_[stack_type]->mapfd_);
```

And for bpf its just an int too:

```
r1 = 0xffff9e6075850000
```

Guess we could change it to a i64* or struct*, for LLVM it doesn't matter but it might remove confusion?